### DISCUSSION TITLE
Full list of UQL functions?

### DISCUSSION THREAD

**Discussion by rogerjbos** [grafana/grafana-infinity-datasource#270] [Q&A] [answered]
In the [UQL](https://sriramajeyam.com/grafana-infinity-datasource/wiki/uql) docs it mentions this section:

```
extend
extend command is similar to project. but instead of selecting the columns, it just adds/replace columns in existing data. extends expects an alias and a function.

parse-json
| extend "dob"=todatetime("dob"), "city"=toupper("city")
following are some of the available functions
```

Then it shows a table of functions.  If that table only lists some of the functions, is there a list somewhere of all the functions available?  I am looking for an if/else type function.

**Comment by yesoreyeram** [comment]
UQL is in very early stage.. no docs yet. But you can see the source code here https://github.com/yesoreyeram/uql

There is no condition support yet. But soon it will be there.

**Reply by rogerjbos** [reply]
Thank you.

**Reply by Harsh130704** [reply]
@yesoreyeram  is it available now ?

**Reply by yesoreyeram** [reply]
https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/uql/

**Comment by yesoreyeram** [comment, marked as answer]
Latest UQL documentation available in https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/uql/

**Comment by rogerjbos** [comment]
Yes, I have access now.  Thanks.

On Wed, Jan 15, 2025 at 4:22 PM Sriram ***@***.***> wrote:

> Latest UQL documentation available in
> https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/uql/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-infinity-datasource/discussions/270#discussioncomment-11847863>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABFSUGKZG6I3C3L3WBTU43L2K3GSNAVCNFSM6AAAAABVHTFZ6GVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOBUG44DMMY>
> .
> You are receiving this because you authored the thread.Message ID:
> <grafana/grafana-infinity-datasource/repo-discussions/270/comments/11847863
> @github.com>
>

-- 
Parity Technologies is a limited company registered in England and Wales 
with registered number 09760015 and registered office at c/o Ignition Law, 
1 Sans Walk, London, England, EC1R 0LT. This message is intended solely for 
the addressee(s) and may contain confidential information. If you have 
received this message in error, please notify us, and immediately and 
permanently delete it. Do not use, copy or disclose the information 
contained in this message or in any attachment. For information about how 
we process data and monitor communications please see our Privacy policy 
(https://www.parity.io/privacy/ <https://www.parity.io/privacy/>) and for 
terms of use please see our Terms of Use policy 
(https://www.parity.io/terms/ <https://www.parity.io/terms/>).
